src/stk/molecular/topology_graphs/polymer/linear/linear.py

```diff
@@ -431,7 +434,11 @@ def _normalize_repeating_unit(repeating_unit):
         base = ord('A')
         return tuple(ord(letter)-base for letter in repeating_unit)
 
-    def _get_building_block_vertices(self, building_blocks, vertices):
+    def _get_building_block_vertices(
+        self,
+        building_blocks: dict[int, BuildingBlock],
+        vertices: Iterable[Vertex],
+    ) -> dict[BuildingBlock, type[Vertex]]:
```

Comment:
this is wrong, it maps a building block to a vertex instance,  not building block to the Vertex class

```python
dict[BuildingBlock, Vertex]
```